includes/StaticDeployJob.php

```diff
@@ -0,0 +1,358 @@
+        $this->save_status_message($message, 'error');
+        return 'cancel';
+    }
+
+    /**
+     * Add a status message about the error and cancel the job
+     * @param  WP_Error $wp_error The error that occurred
+     * @return void
+     */
+    protected function error_occurred($wp_error) {
+        $errorMessage = is_a($wp_error, WP_Error::class)
+            ? $wp_error->get_error_message()
+            : $wp_error->getMessage();
+
+        Util::debug_log("An error occurred: " . $errorMessage);
+        Util::debug_log($wp_error);
+        $message = sprintf(__("An error occurred: %s", 'simply-static'), $errorMessage);
+        $this->save_status_message($message, 'error');
+        return 'cancel';
+    }
+
+
+    /**
+     * Shutdown handler for fatal error reporting
+     * @return void
+     */
+    public function shutdown_handler() {
+        // Note: this function must be public in order to function properly.
+        $error = error_get_last();
+        // only trigger on actual errors, not warnings or notices
+        if ($error && in_array($error['type'], array(E_ERROR, E_CORE_ERROR, E_USER_ERROR))) {
+            $this->clear_scheduled_event();
+            $this->unlock_process();
+            $this->cancel_process();
+
+            $end_time = Util::formatted_datetime();
+            $this->options
+                ->set('archive_end_time', $end_time)
+                ->save();
+
+            $error_message = '(' . $error['type'] . ') ' . $error['message'];
+            $error_message .= ' in <b>' . $error['file'] . '</b>';
+            $error_message .= ' on line <b>' . $error['line'] . '</b>';
+
+            $message = sprintf(__("Error: %s", 'simply-static'), $error_message);
+            Util::debug_log($message);
+            $this->save_status_message($message, 'error');
+
+            
+            // restore initial options
+            $restoreInitialOptionsTask = new RestoreInitialOptionsTask();
+            $restoreInitialOptionsTask->perform();
+        }
+    }
+
+    protected function compose_task_list() {
+        // check if we are working on a single deploy
+        $single_deploy_id = get_option(Plugin::SLUG . '_single_deploy_id');
+        if ($single_deploy_id) {
+            $task_list = [
+                'store_initial_options',
+                'setup_single',
+                'fetch_urls',
+                'transfer_files_locally',
+                'wrapup',
+                'restore_initial_options',
+                'sync',
+                'invalidate',
+            ];
+        } else {
+            $task_list = [
+                'store_initial_options',
+                'setup',
+                'fetch_urls',
+                'transfer_files_locally',
+                'wrapup',
+                'restore_initial_options',
+                'sync',
+                'invalidate',
+            ];
+        }
```

Comment:
Do you think it's more clear in this function to only replace the `setup` task with a ternary statement? It saves repeating the entire task list.